### PR TITLE
Add support for 4th level TLDs.

### PR DESCRIPTION
**Pull request by jbaggs**
https://publicsuffix.org/list now contains 4th level TLDs. e.g: `s3-website.eu-central-1.amazonaws.com`

### CODE REVIEW

File: scripts/main.bro [hunk 1]
```diff
@@ -69,6 +71,8 @@ function zone_by_depth(domain: string, depth: count): string
 function effective_tld(domain: string): string
 	{
 	local depth=1;
+	if ( effective_tlds_4th_level in domain )
+		depth=4;
 	if ( effective_tlds_3rd_level in domain )
```
Comment (sethhall) [inline scripts/main.bro]: I suspect this should be an else-if now that you added the additional if statement.  Agree?

```suggestion
	else if ( effective_tlds_3rd_level in domain )
```

**Review by sethhall — CHANGES_REQUESTED** [review]
Just a tiny change.

**Comment by sethhall** [comment]
Oh, could you regenerate the tld-data.bro script too?

File: scripts/main.bro (hunk 1 above)
Comment (jbaggs) [reply scripts/main.bro]: Oh, yes. That makes sense.  It should be an else-if at line 88 as well. I'm not really familiar with this new inline suggestion mechanism either, but it appears to not allow further changes while reviewing.

File: scripts/main.bro [hunk 2]
```diff
@@ -79,6 +83,8 @@ function effective_tld(domain: string): string
 function effective_domain(domain: string): string
 	{
 	local depth=2;
+	if ( effective_tlds_4th_level in domain )
+		depth=5;
 	if ( effective_tlds_3rd_level in domain )
```
Comment (jbaggs) [inline scripts/main.bro]:
```suggestion
	else if ( effective_tlds_3rd_level in domain )
```

Comment (sethhall) [inline scripts/main.bro]: Whoops, there was one more.

```suggestion
	if ( effective_tlds_3rd_level in domain )
```

Comment (sethhall) [inline scripts/main.bro]: Could you also update the domain-tld.bro script?

**Review by sethhall — CHANGES_REQUESTED** [review]

File: scripts/main.bro (hunk 2 above)
Comment (jbaggs) [reply scripts/main.bro]: Looks like we were both suggesting the change at 88, but I take it mine did not show up on your end. I think there was maybe one more step in committing the suggestion. 

The tld-data.bro script?

Comment (jbaggs) [reply scripts/main.bro]: I'm wondering if we're hitting the limits of "suggested changes" here. I had made a suggested change for line 88 to add else-if there as well. Then you appeared to have caught that change was needed independently, and suggested the same change. Now I'm seeing something different in your suggestion, but I think my repo is in sync as far as the actual changes go. Does it look like that to you?

**Comment by sethhall** [comment]
Haha, two of us cluelessly stumbling through this new feature. :)

We got it done.  Thanks for spending the time!

**Comment by jbaggs** [comment]
Yes. Good to now know how it functions, at least. Thanks for working through the changes Seth.

**Pull request merged**